docs/articles/documentation/test-api/obtaining-data-from-the-client/examples-of-using-client-functions.md

```diff
@@ -0,0 +1,174 @@
+        return $('div.column.col-2').text();
+    });
+
+    const text = await clientFunction();
+    console.log(text);
+});
+```
+
+## Accessing Child Nodes in the DOM Hierarchy
+
+To find all child elements (not [nodes](https://developer.mozilla.org/en-US/docs/Web/API/Node)) of all nodes within a DOM hierarchy, TestCafe provides the [Selector.child](../selecting-page-elements/selectors/functional-style-selectors.md#child) method. However, you may need to access exactly child nodes of some node. You can do this by using a client function.
+
+Suppose that you test the *example.html* page with the following HTML code.
+
+```js
+<!DOCTYPE html>
+<html>
+<body id="testedpage">
+This is my tested page. <!--This is the first child node of <body>-->
+<p>My first paragraph.</p>
+<p>My second paragraph.</p>
+</body>
+</html>
+```
+
+Also suppose that you need to obtain the text *This is my tested page*. This text is considered as a child node, not a child element of the body element. So, you can’t access this text by using `Selector.child`, but can do this by obtaining the [childNodes](https://developer.mozilla.org/en-US/docs/Web/API/Node/childNodes) property from the client side.
+
+```js
+import { Selector } from 'testcafe';
+import { ClientFunction } from 'testcafe';
+
+fixture `My Fixture`
+   .page `examplepage.html`;
+
+   const getChildNodeText = ClientFunction(index => {
+       return getEl().childNodes[index].textContent;
+   },{ dependencies: { getEl: Selector('#testedpage') } });
+
+
+test('My Test', async t => {
+   console.log(await getChildNodeText(0));
+});
+});
+```
+
+## Complex DOM Queries
+
+Client functions can be helpful for complex DOM queries. For example, a tested page contains a table with some data and you need to validate data only from particular two columns. To do this, you can obtain data from these columns, push them to an array and then compare the returned array with the expected one.
+
+Below is a sample test for the [https://js.devexpress.com/](https://js.devexpress.com/) page that contains a grid. The sample demonstrates how to create a client function that returns an array containing data from the grid’s **Customer** and **Sale Amount** columns.
+
+```js
+import { ClientFunction} from 'testcafe';
+
+fixture `Get sale amount`
+   .page('https://js.devexpress.com/');
+
+   const getSaleAmount = ClientFunction(() => {
+       const elements = document.querySelector('.dx-datagrid-rowsview').querySelectorAll('td:nth-child(3),td:nth-child(7)');
+       const array = [];
+
+       for (const i = 0; i <= elements.length - 2; i+=2) {
+           const saleAmount  = elements[i+1].textContent;
+           const customerName = elements[i].textContent;
+
+           if (customerName && saleAmount)
+               array.push(`Customer ${elements[i+1].textContent}: ${elements[i].textContent}`);
+       }
+
+       return array;
+   });
+
+   test('Test1', async t => {
+   console.log(await getSaleAmount());
```

Comment:
I don't sure in this case, but I still prefer real assertions

Reply:
I don't understand what you mean.

Reply:
Discussed in private